### DISCUSSION TITLE
How to Generate Parts List from Model

### DISCUSSION THREAD

**Discussion by dieterjanssen** [leozide/leocad#771] [Q&A]
Within LeoCAD, once a model is complete, how do you generate a list of the parts used in the project?
![LEGO Edith Farnsworth 01a](https://user-images.githubusercontent.com/92032646/151810232-6f62644d-f43d-4c4f-98e8-c59cfa49b00c.png)

**Comment by nathaneltitane** [comment]
![image](https://user-images.githubusercontent.com/5674757/152035173-f3c41710-d16c-4298-a388-e27b28b63aba.png)

**Comment by nathaneltitane** [comment]
![image](https://user-images.githubusercontent.com/5674757/152035264-b8d0504c-19e0-465a-ab7b-54598a3722e6.png)
![image](https://user-images.githubusercontent.com/5674757/152035289-3340e5fe-185e-4d0a-b1ce-2ffbb36b8652.png)

would be nice if @leozide could add an export button that links to the same functions into this dialog

**Reply by leozide** [reply]
It depends on how you want the list, do you want a BOM, a table with images or text?

**Reply by dieterjanssen** [reply]
A table with images would be ideal, but any list would work.

**Reply by leozide** [reply]
Try File->Export->HTML, you'll get something like this:

![image](https://user-images.githubusercontent.com/9685066/152696635-aae2bfda-6ee3-4863-a21b-031c78c11b3b.png)

**Reply by dieterjanssen** [reply]
Thanks Leozide & nathaneltitane - that's definitely a step in the right direction. Would you know if it'll also list part numbers? The plan would be to eventually purchase the parts to confirm this actually works. And any suggestions for how-to's on setting up steps would be welcome as well.
![LEGO-Edith-Farnsworth-ldr-Step-1](https://user-images.githubusercontent.com/92032646/153014283-a1bc04dc-59d9-4557-ab5e-c3eb76bf32b1.png)
![LEGO Edith Farnsworth 06](https://user-images.githubusercontent.com/92032646/153019010-ea60f783-b49a-4629-8f6c-143221ad4eeb.png)

**Comment by nathaneltitane** [comment]
@dieterjanssen you want to use this:

![image](https://user-images.githubusercontent.com/5674757/153025199-5e437463-39fa-46c2-9e73-a7a23f5f6548.png)

**Reply by dieterjanssen** [reply]
Thanks @nathaneltitane - it's creating a visual list, but doesn't have part numbers. That said, the CSV option includes all that info. It be great to have the visual with that info.
![image](https://user-images.githubusercontent.com/92032646/153439910-f800cf01-8a62-4641-a854-c4890281bfc7.png)

**Reply by nathaneltitane** [reply]
> Thanks @nathaneltitane - it's creating a visual list, but doesn't have part numbers. That said, the CSV option includes all that info. It be great to have the visual with that info. ![image](https://user-images.githubusercontent.com/92032646/153439910-f800cf01-8a62-4641-a854-c4890281bfc7.png)

My pleasure... 

Come this point, it would be up to @leozide to decide whether or not he wishes to add options to enable thumbnails or not to this feature or other similar ones.

This goes back to a few discussions and tickets that were open that suggest adding options, with checkbox toggles and the corresponding keyboard combination entries (often incomplete or not functional in some cases) in the preferences menu to let users tailor their experience of leocad depending on what they would like to see, generate or render, and would also help with optimizing leocad for various system types as well.

**Comment by MrEdgeNL** [comment]
@dieterjanssen nice building, a bit off topic: could you provide the part numbers for the vertical-used H-shaped Lego Parts?
I cann't find them on the internet sofar..
